### DISCUSSION TITLE
Correct way to implement a collapsing Nav?

### DISCUSSION THREAD

**Discussion by RichiCoder1** [ariakit/ariakit#3182] [Questions] [answered]
I'm implementing a navbar for a site I'm working and am working on making it mobile friendly. Have about 5-6 items, with a few using submenus. Currently using MenuBar (as described in the Nav menu example and it's working well).

Was wondering what the most correct way to implement the mobile version was? Currently just hiding the desktop via (via tailwind `md:`) and then using `Disclosure` w/ a standard hamburger + an absolutely positioned `DisclosureContent` to contain the mobile version. Is this alright or should I be using something like a `Dialog` to get things like backdrop and click backdrop to dismiss?

**Comment by diegohaz** [comment, marked as answer]
It's better to use `Dialog`, not just due to the backdrop and such but also because of the auto-focus feature. This is important for people using a screen reader on mobile.

**Reply by RichiCoder1** [reply]
@diegohaz awesome, I'll give that a go! Would I want to use a standard button here for the toggle (with title) or would disclosure still make sense?

**Reply by diegohaz** [reply]
You can use a normal button.

**Reply by RichiCoder1** [reply]
![MobileMenu](https://github.com/ariakit/ariakit/assets/2391878/1f1e7179-055a-42c6-ae25-2ff8840b990b)
Working great! Ty again
